js/server/modules/@arangodb/foxx/manager.js

```diff
@@ -637,8 +685,17 @@ function _deleteServiceFromPath (mount, force) {
 // FoxxService.basePath(mount) and FoxxService.bundlePath(mount). This is to
 // avoid a race condition with selfHeal() which could delete the files before
 // the service is saved to the database.
-function _install (tempService, tempBundlePath, options = {}) {
+function _install(tempService, tempBundlePath, options = {}) {
+
+  const locked = _checkServiceModificationLock();
+  if (locked) {
+    // TODO: Throw if we were not able to get the lock within 10 seconds
+    return;
+  }
+
   try {
+    _setServiceModificationLock(true);
```

Comment:
Two installs on the same coordinator at the same time could cause a race condition here:

Thread A checks the lock, sees it's free and proceeds.
Thread B checks the lock, also sees it's free and proceeds.
Thread A and B can now set the lock in any order.
If thread A or B releases the lock that doesn't mean the other thread is done too so this race condition can escalate further.